pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -568,7 +568,10 @@ type KafkaTrigger struct {
 	// Name of the topic.
 	// More info at https://kafka.apache.org/documentation/#intro_topics
 	Topic string `json:"topic" protobuf:"bytes,2,opt,name=topic"`
+	// +optional
 	// Partition to write data to.
+	// By default the value is “0” to use random partition you need set “-1”  more info at "https://www.confluent.io/blog/apache-kafka-producer-improvements-sticky-partitioner/
```

Comment:
What about making it `*int32`?

Reply:
This blog is unrelated to the general concept of Kafka partitioning.

The sticky partitioner is specific to the Java implementation